util/rbac/rbac.go

```diff
@@ -74,6 +83,16 @@ func (e *Enforcer) SetDefaultRole(roleName string) {
 	e.defaultRole = roleName
 }
 
+// SetDefaultRole sets a default role to use during enforcement. Will fall back to this role if
+// normal enforcement fails
+func (e *Enforcer) SetScopes(scopes []string) {
+	if len(scopes) == 0 {
```

Comment:
I think we are not supporting the difference between: omitting scopes from the configmap vs. setting an empty scopes list. The difference is important because someone may want the following config:

```yaml
data:
  scopes: []
```

Which is what is needed for Google OAuth. Google OAuth does not have an equivalent `groups` scope. It also not tolerant to the caller requesting the `groups` scope. So user needs a way to both (a) prevent requesting the groups scope, and (b) prevent enforcement of the groups scope.

Reply:
You are right. Fixed